perun-cli/Perun/beans/ApplicationFormItem.pm

```diff
@@ -0,0 +1,313 @@
+        $i18n = $self->{_i18n};
+    } else {
+        $i18n = undef;
+    }
+
+    return { id                   => $id,
+        shortname                 => $shortname,
+        required                  => $required,
+        type                      => $type,
+        federationAttribute       => $federationAttribute,
+        perunSourceAttribute      => $perunSourceAttribute,
+        perunDestinationAttribute => $perunDestinationAttribute,
+        regex                     => $regex,
+        applicationTypes          => \@applicationTypes,
+        ordnum                    => $ordnum,
+        forDelete                 => $forDelete,
+        i18n                      => $i18n,
+        beanName                  => 'ApplicationFormItem'
+    };
+}
+
+sub getId
+{
+    my $self = shift;
+
+    return $self->{_id};
+}
+
+sub getShortname
+{
+    my $self = shift;
+
+    return $self->{_shortname};
+}
+
+sub setShortname
+{
+    my $self = shift;
+    $self->{_shortname} = shift;
+
+    return;
+}
+
+sub isRequired
+{
+    my $self = shift;
+    return ($self->{_required}) ? 'true' : 'false';
+}
+
+sub setRequired
+{
+    my $self = shift;
+    my $value = shift;
+    if (ref $value eq "JSON::XS::Boolean")
+    {
+        $self->{_required} = $value;
+    } elsif ($value eq 'true' || $value eq 1)
+    {
+        $self->{_required} = JSON::XS::true;
+    } else
+    {
+        $self->{_required} = JSON::XS::false;
+    }
+
+    return;
+}
+
+sub getType
+{
+    my $self = shift;
+
+    return $self->{_type};
+}
+
+sub setType
+{
+    my $self = shift;
+    $self->{_type} = shift;
+
+    return;
+}
+
+sub getFederationAttribute
+{
+    my $self = shift;
+
+    return $self->{_federationAttribute};
+}
+
+sub setFederationAttribute
+{
+    my $self = shift;
+    $self->{_federationAttribute} = shift;
+
+    return;
+}
+
+sub getPerunSourceAttribute
+{
+    my $self = shift;
+
+    return $self->{_perunSourceAttribute};
+}
+
+sub setPerunSourceAttribute
+{
+    my $self = shift;
+    $self->{_perunSourceAttribute} = shift;
+
+    return;
+}
+
+sub getPerunDestinationAttribute
+{
+    my $self = shift;
+
+    return $self->{_perunDestinationAttribute};
+}
+
+sub setPerunDestinationAttribute
+{
+    my $self = shift;
+    $self->{_perunDestinationAttribute} = shift;
+
+    return;
+}
+
+sub getRegex
+{
+    my $self = shift;
+
+    return $self->{_regex};
+}
+
+sub setRegex
+{
+    my $self = shift;
+    $self->{_regex} = shift;
+
+    return;
+}
+
+sub getApplicationTypes
+{
+    my $self = shift;
+
+    return @{$self->{_applicationTypes}};
+}
+
+sub setApplicationTypes
+{
+    my $self = shift;
+    $self->{_applicationTypes} = shift;
+
+    return;
+}
+
+sub getOrdnum
+{
+    my $self = shift;
+
+    return $self->{_ordnum};
+}
+
+sub setOrdnum
+{
+    my $self = shift;
+    $self->{_ordnum} = shift;
+
+    return;
+}
+
+sub isForDelete
+{
+    my $self = shift;
+
+    return ($self->{_forDelete}) ? 'true' : 'false';
```

Comment:
Depending on your usage you might want to retrieve original value (boolean or 0/1) which you set using ``setForDelete``. This kind of handling is useful for printing the value in tables in a consistent way. You can have ``sub isForDeleteToPrint`` for returning true/false as strings.